cmd/metal-api/internal/fsm/fsm.go

```diff
@@ -166,125 +24,70 @@ func HandleProvisioningEvent(log *zap.SugaredLogger, ec *metal.ProvisioningEvent
 		return nil, fmt.Errorf("provisioning event must not be nil")
 	}
 
-	clone := *ec
-	container := &clone
-
-	container, err := checkProvisioningEvent(event, container, log)
-	if err != nil {
-		return nil, fmt.Errorf("internal error while calculating provisioning event container for machine %s", container.ID)
-	}
-
-	return container, nil
-}
+	var (
+		clone       = *ec
+		container   = &clone
+		latestEvent = func() string {
+			if len(container.Events) != 0 {
+				return container.Events[0].Event.String()
+			}
+			return ""
+		}()
```

Comment:
```suggestion
		latestEvent = len(container.Events) != 0 ? container.Events[0].Event.String() : ""
```

What do you think about this syntax?

Reply:
Put into separate function.